### DISCUSSION TITLE
Loading tiled images from the filesystem

### DISCUSSION THREAD

**Discussion by abiswas97** [openseadragon/openseadragon#2400] [Q&A]
Hi,

I'm currently exploring the potential/feasibility of loading a DZI and it's tiles on demand from the local filesystem. From an initial search, it looks like I can use the current [File System Access API](https://developer.chrome.com/articles/file-system-access/#opening-a-directory-and-enumerating-its-contents) to gain access to a folder, and use a directory access handler to get access to my tiles on demand. 

For loading this into OSD, my initial idea was to load the tile and then call `URL.createObjectURL()`, and return the resulting URL to `getTileUrl`. However, createObjectURL comes with the added need for revoking each URL once consumed, and I was wondering if it's possible to pass the tile data directly to OSD instead of passing the tileURL? 

Would also love to hear if anyone has taken a crack at this before, and if there are any pitfalls I should be wary of.

**Comment by iangilman** [comment]
Interesting scenario! This sounds like a job for: https://openseadragon.github.io/examples/advanced-data-model/

@aiosa may have additional thoughts.

**Reply by abiswas97** [reply]
Looks like I've really been missing out by not upgrading to OSD 4.0! 

@pearcetm provided just what I needed in the [issue ](https://github.com/openseadragon/openseadragon/issues/2403#issuecomment-1705470000) I opened about this. I'm planning to create a fork of geotiff tilesource or a new plugin inspired by it for handling local DZI tilesources.

**Reply by iangilman** [reply]
Excellent :)
